File: week09/week09-3.py
class Solution:
    def setZeroes(self, a: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        row=[]
        col=[]
        n1=len(a)
        n2=len(a[0])
        for i in range(n1):
            for j in range(n2):
                if a[i][j]==0:
                    row.append(i)
                    col.append(j)
        for i in range(n1):
            for j in range(n2):
                if i in row or j in col:a[i][j]=0
        return a

File: week09/week09-3.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int m = matrix.size(), n = matrix[0].size();
        vector<int> rows(m,1), cols(n,1);  //row(m,1) 新增大小為m*n的空間，假設一開始都是1
        for(int i=0; i<m; i++){
            for(int j=0; j<n; j++){
                if(matrix[i][j]==0){//如果遇到0
                    rows[i]=0;//紀錄第幾行有沒有0
                    cols[j]=0;//紀錄第幾列有沒有0
                }
            }
        }
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(rows[i]==0 || cols[j]==0){//如果這行或這列的某個值為0
                    matrix[i][j]=0;//則這一格變成0
                }
            }
        }
    }
};